sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/TimeWindowFilterSettings.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.feature.management.implementation.timewindow;
+
+import com.azure.spring.cloud.feature.management.implementation.timewindow.recurrence.Recurrence;
+import org.springframework.util.StringUtils;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class TimeWindowFilterSettings {
+    /**
+     * An optional start time used to determine when a feature should be enabled.
+     * If no start time is specified the time window is considered to have already started.
+     * */
+    private ZonedDateTime start;
+    /**
+     * An optional end time used to determine when a feature should be disabled.
+     * If no end time is specified the time window is considered to never end.
+     * */
+    private ZonedDateTime end;
+    /**
+     * Add-on recurrence rule allows the time window defined by Start and End to recur.
+     * The rule specifies both how often the time window repeats and for how long.
+     * */
+    private Recurrence recurrence;
+
+    /**
+     * @param startTime the start time to determine when a feature should be enabled
+     * */
+    public void setStart(String startTime) {
+        try {
+            this.start = StringUtils.hasText(startTime)
```

Comment:
Can we split this out into a utility class. We have the same code a few times here, and I expect we might end up using it in another class in the future.

Reply:
Yes, extract the same logic to a seprate function.